### PR TITLE
US #9: Feat violation alert message

### PR DESCRIPTION
**Pull request by nathanss**
This covers [user story #9](https://trello.com/c/6SyA7sFe/33-us-9-alerta-de-viola%C3%A7%C3%A3o-de-requisito).

Creates a cron job that runs every day 9AM to check for open/pending requirements that will expire in a week or less.

To test, follow the suggestion described in line 5 of cronJobs.js file. Change the line to make it run every minute.

Check the user story for the acceptance criteria.

### CODE REVIEW

File: routes/projectRequirements.js [hunk 1]
```diff
@@ -75,6 +78,7 @@ router.post('/update', async function(req, res) {
   const requirementName = await getOneEntryFromCollection(db, "requirementcollection", {_id: projectRequirement.requirement});
   const projectName = await getOneEntryFromCollection(db, "projectcollection", {_id: projectRequirement.proj});
   const userName = await getOneEntryFromCollection(db, "usercollection", {_id: projectRequirement.responsibleUser});
+  projectRequirement.solvedate = new Date(projectRequirement.solvedate);
```
Comment (isaqueha) [inline routes/projectRequirements.js:82]: It's ok in the `/update` route.
The `/add` route also needs this setup.

File: cronJobs.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                $eq: "PENDING"
+              }
+            }
+          ]
+        }
+      ]
+    });
+  
+    if (requirementsAboutToExpire && requirementsAboutToExpire.length) {
+      const messages = requirementsAboutToExpire.map((a) => {
```
Comment (isaqueha) [inline cronJobs.js]: Can you make a method to return the messages formatted?

File: cronJobs.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        return {
+          messagetext: `Requirement ${a.requirementName} of project ${a.projectName} is in status ${a.status} and will expire in ${moment.utc(a.solvedate).format("DD-MM-YYYY")}. Please expedite.`,
+          sender: "SYSTEM",
+          receivers: a.responsibleUser,
```
Comment (isaqueha) [inline cronJobs.js]: Maybe you need to send to the Project Manager as well.

> O Gerente de Projetos deve receber os alertas dos projetos em que está vinculado

File: cronJobs.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+var CronJob = require('cron').CronJob;
+var moment = require('moment');
+
+module.exports = async function(db) {
+  new CronJob('0 9 * * *', async function() { // replace with * * * * * to run every minute for testing
+    const now = new Date();
+    const weekFromNow = new Date();
+    weekFromNow.setDate(now.getDate() + 7);
+    const requirementsAboutToExpire = await db.get('projectRequirementsCollection').find({
+      $and: [
```
Comment (isaqueha) [inline cronJobs.js:10]: Can you make a method to get this query?

**Review by isaqueha — CHANGES_REQUESTED** [review]

**Comment by isaqueha** [comment]
@nathanss, this feature is Noice
image

File: cronJobs.js (hunk 4 above)
Comment (nathanss) [reply cronJobs.js:10]: y

File: cronJobs.js (hunk 2 above)
Comment (nathanss) [reply cronJobs.js]: y

File: cronJobs.js (hunk 3 above)
Comment (isaqueha) [reply cronJobs.js]: We are going to let this out for now

File: cronJobs.js (hunk 4 above)
Comment (isaqueha) [reply cronJobs.js:10]: We are going to let this out for now

**Review by isaqueha — APPROVED** [review]

**Pull request merged**